cloud/services/virtualmachines/virtualmachines.go

```diff
@@ -156,19 +156,9 @@ func (s *Service) Reconcile(ctx context.Context) error {
 					Type: compute.ResourceIdentityTypeSystemAssigned,
 				}
 			} else if vmSpec.Identity == infrav1.VMIdentityUserAssigned {
-				if len(vmSpec.UserAssignedIdentities) == 0 {
```

Comment:
I refactored this, and moved it to converters [here](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/47d4027e55b281e23b62aa7a87df8dabad479727/cloud/converters/identity.go#L33). Since the logic for vmss is identical to vm, I thought it would be easier to read. Let me know, if it's okay.

Reply:
oh ok, got it, sounds good, I will run this template a bit and test it out before putting in more comments